dht.go

```diff
@@ -507,3 +578,46 @@ func (dht *IpfsDHT) newContextWithLocalTags(ctx context.Context, extraTags ...ta
 	) // ignoring error as it is unrelated to the actual function of this code.
 	return ctx
 }
+
+func (dht *IpfsDHT) handleProtocolChanges(ctx context.Context) {
+	// register for event bus protocol ID changes
+	sub, err := dht.host.EventBus().Subscribe(new(event.EvtPeerProtocolsUpdated))
+	if err != nil {
+		panic(err)
+	}
+	defer sub.Close()
+
+	pmap := make(map[protocol.ID]bool)
+	for _, p := range dht.protocols {
+		pmap[p] = true
+	}
+
+	for {
+		select {
+		case ie, ok := <-sub.Out():
+			e, ok := ie.(event.EvtPeerProtocolsUpdated)
+			if !ok {
+				logger.Errorf("got wrong type from subscription: %T", ie)
+				return
+			}
+
+			protos, err := dht.peerstore.SupportsProtocols(e.Peer, dht.protocolStrs()...)
+			if err != nil {
+				continue
+			}
+
+			if len(protos) > 0 {
+				for _, p := range e.Added {
+					if pmap[p] {
+						dht.routingTable.Update(e.Peer)
+						break
+					}
+				}
+			} else {
+				dht.routingTable.Remove(e.Peer)
+			}
+		case <-ctx.Done():
+			return
```

Comment:
Can't we use a go-process for this routine and proc.Closing here as that's what we are doing everywhere else too ?

Reply:
sure, we can do a goprocess purge in a separate PR

Reply:
Cool. I pointed this out because we don't do a proper shutdown of this ctx when we close the DHT right now. But, we have a PR open for that now.